### ISSUE TITLE
[Accessibility]: Show as table button in charts does not announce dialog being open

### ISSUE THREAD

**Issue by amortiz** [carbon-design-system/carbon-charts#2038]
### Application/Team

IBM Cloud infrastructure

### Environment

Browser

### What happened? What did you expect to happen instead?

If you click  "Show as table" button on any chart, a screen reader does not announce that the modal is open, and focus is not anywhere on the dialog. I think if you were to focus on the close button or "Download as CSV" button on the modal, it would comply.

Full AVT report:

```
Issue description: When SR user open show Table button,a dialog is opened . No commutation is provided for user a Dailog is opened.

Recommended fix: SR reader should Convey the user that a new Dialog is oped

User impact: When the name, role or state of an element is not conveyed appropriately, it prevents screen reader users to interact smoothly with the page element.

WCAG Mapping: 4.1.2 Name, Role, Value

Level of Compliance: A

Severity: Medium

Test tool: Keyboard_Chrome

```

<img width="980" height="553" alt="Image" src="https://github.com/user-attachments/assets/01feaafe-f4b9-48a0-a8a4-1a2ae3e36e8c" />

### What WCAG 2.1 checkpoint does the issue violate?

4.1.2 Name, Role, Value

### Version

Latest, just try any chart

### Data & options used

```js

```

### Relevant log output

```js

```

### What priority level would this issue be in your opinion?

P3
